Source/PagesController.swift

```diff
@@ -148,5 +132,23 @@ import UIKit
       }
     }
   }
+}
+
+extension Array {
+  func at(index: Int?) -> T? {
+    if let index = index where index >= 0 && index < self.endIndex {
+      return self[index]
+    } else {
+      return nil
+    }
+  }
+}
 
+func nextIndex(x: Int?) -> Int? {
+  return x.map { $0 + 1}
```

Comment:
did you get what it does ?
map for optionals unwrap it and apply map body if value exist. it not return empty optional